Add UI tests for QML

acolombier opened the pull request:
This is a naive attempt to address #12266

Note that two commits were also added to address low hanging fruits of #14297 and reduce amount of compute used by our our CI

Note: I was unable to get it running on Mac - I'll see if I can get something but this will likely have to be skipped at first since I don't own a Mac and have limited interest in supporting Apple ecosystem

Depends on https://github.com/mixxxdj/vcpkg/pull/171

Disabling on Windows 11 arm as the runner VM appears not to be correctly setup.
<img width="400" alt="image" src="https://github.com/user-attachments/assets/92345ab1-c280-4b42-b9e2-e3acc9933197" />

UI results will be visible in the job summary. I couldnt get it to work in the comment for now!
<img width="600" alt="image" src="https://github.com/user-attachments/assets/0f258f03-973e-4b0e-84d8-322f3118b1b5" />

Eve00000 commented:
>     * Record/publish video?

Yes, and we can test on Mac VM.

Eve00000 commented:
> Do you mean manual testing?

we need to start somewhere

JoergAtGithub commented:
Did you have a look at: https://github.com/faaxm/spix
I've no experience with it, but it seems to me pretty much what is needed here.

JoergAtGithub commented on tools/windows_buildenv.bat:
```diff
@@ -29,8 +29,8 @@ IF DEFINED BUILDENV_RELEASE (
     SET BUILDENV_BRANCH=2.6
     SET VCPKG_TARGET_TRIPLET=x64-windows
-    SET BUILDENV_NAME=mixxx-deps-2.6-x64-windows-12239ed
-    SET BUILDENV_SHA256=5c60b2c61d6448a99979d7cc997e92f2fd5b4b65f65e2439abfca3fa6fd30f8d
+    SET BUILDENV_NAME=mixxx-deps-2.6-x64-windows-7cc6152
+    SET BUILDENV_SHA256=44b0fc493959ffec09dc48af3b4400d3e550aed970187222f95558942a42e066
```
In CI we use the release only version of the buildenv. The full version with additional debug binaries is for local development. This change will not enable ffmpeg therfore.

acolombier replied:
Yes, I've commented that part out in the CI so I can start trying and see if I can get ffmpeg to work before submitting a PR to make a new vcpkg rel version

acolombier commented:
@JoergAtGithub I have resync'ed this branch and updated it with the latest module refactor from Spix - good news is, it is working great... and detected the bug, fixed by #15361.

Video results can be found in the [Artifacts sections](https://github.com/mixxxdj/mixxx/actions/runs/18935253747?pr=14360) - I'll spend some time to add a nice auto-comments with links to video recording, but in the meantime, it would be good to merge this other PR so we can fix the tests

acolombier commented:
A bit of back and forth, but we should now have something stable. I had to do some changes on Android setup, so not sure if you want me to split the commits?

JoergAtGithub commented:
There seems to be quite a lot of changes in CI setup. Would it make sense to move them in another PR?

acolombier committed:
fix(rendergraph): disable QShader for GL on Qt 6.6

acolombier committed:
feat: add QML UiTest

acolombier committed:
chore(test): add a basic UI test

acolombier committed:
fix(library): remove library folder prompt on empty source list if in developer mode

acolombier commented:
Friendly ping @JoergAtGithub

JoergAtGithub reviewed (changes requested):
- Please move the changes for the Coverage build to an own PR, as it seems unrelated to the QML UI tests. Combining unrelated changes into a single PR, is a practice that often leads to delays in approval.
- Please check the description of the PR. I think what is written there is not correct/up-to-date.

acolombier committed:
fixup! feat: add QML UiTest
